src/lib/storage/dictionary_segment/dictionary_encoder.hpp

```diff
@@ -28,112 +29,92 @@ class DictionaryEncoder : public SegmentEncoder<DictionaryEncoder<Encoding>> {
   static constexpr auto _uses_vector_compression = true;  // see base_segment_encoder.hpp for details
 
   template <typename T>
-  std::shared_ptr<BaseEncodedSegment> _on_encode(const std::shared_ptr<const ValueSegment<T>>& value_segment) {
-    // See: https://goo.gl/MCM5rr
-    // Create dictionary (enforce uniqueness and sorting)
-    const auto& values = value_segment->values();
+  std::shared_ptr<BaseEncodedSegment> _on_encode(const AnySegmentIterable<T> segment_iterable,
+                                                 const PolymorphicAllocator<T>& allocator) {
+    // Vectors to gather the input segment's data. This data is used in a later step to
+    // construct the actual dictionary and attribute vector.
+    std::vector<T> values;          // does contain the actual values (no NULLs)
```

Comment:
Simply “contains”? Also, just to make sure no one misses it “This is different from other definitions of `values`”?

Reply:
Changed to "contains". I guess you mean that other times `values` stores NULL values as well? I think using `vector<T>` makes that clear enough. Or do you mean something else?

Reply:
(Almost?) everywhere else, this vector stores `T{}` for NULL values. Otherwise, we wouldn't be able to access values by index.

Here, it is fine, because we never access by index. I'd even go as far as to call it `dense_values`